lang/java/reef-bridge-proto-java/src/main/java/org/apache/reef/bridge/Launch.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.bridge;
+
+import com.google.common.collect.Lists;
+import org.apache.reef.bridge.grpc.GRPCDriverBridgeService;
+import org.apache.reef.bridge.parameters.BridgeDriverProcessCommand;
+import org.apache.reef.bridge.parameters.DriverServicePortRangeCount;
+import org.apache.reef.bridge.parameters.DriverServiceStartPort;
+import org.apache.reef.client.DriverConfiguration;
+import org.apache.reef.client.DriverLauncher;
+import org.apache.reef.runtime.local.client.LocalRuntimeConfiguration;
+import org.apache.reef.runtime.yarn.client.YarnClientConfiguration;
+import org.apache.reef.tang.*;
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.annotations.NamedParameter;
+import org.apache.reef.tang.exceptions.BindException;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.tang.formats.CommandLine;
+import org.apache.reef.tang.formats.ConfigurationModule;
+import org.apache.reef.util.EnvironmentUtils;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Pool of Evaluators example - main class.
+ */
+public final class Launch {
```

Comment:
Where is this code run?

Reply:
Client submission of the job.

Reply:
Is this the "Client" in the sense of the machine that submits the job? Why would that be a process of its own as opposed to a library?

Reply:
Yes. 
Because anything non-java (e.g., C#) would not be able to call into it.